Sming/SmingCore/Platform/Station.cpp

```diff
@@ -388,6 +388,79 @@ void StationClass::smartConfigStop() {
 	smartConfigCallback = NULL;
 }
 
+#ifdef ENABLE_WPS
+void StationClass::internalWpsConfig(wps_cb_status status)
+{
+	bool processInternal=true;
+	if (wpsConfigCallback){
+		processInternal=wpsConfigCallback(status);
+	}
+	if (processInternal){
+		switch (status) {
+			case WPS_CB_ST_SUCCESS:
+				debugf("wifi_wps_status_cb(): WPS_CB_ST_SUCCESS\n");
+				wpsConfigStop();
+				connect();
+				break;
+			case WPS_CB_ST_FAILED:
+				debugf("wifi_wps_status_cb(): WPS_CB_ST_FAILED\n");
+				wpsConfigStop();
+				connect();	// try to reconnect with old config
+				break;
+			case WPS_CB_ST_TIMEOUT:
+				debugf("wifi_wps_status_cb(): WPS_CB_ST_TIMEOUT\n");
+				wpsConfigStop();
+				connect();	// try to reconnect with old config
+				break;
+			case WPS_CB_ST_WEP:
+				debugf("wifi_wps_status_cb(): WPS_CB_ST_WEP\n");
+				break;
+			default :
+				debugf("wifi_wps_status_cb(): unknown wps_cb_status %d\n",status);
+				wpsConfigStop();
+				connect();   // try to reconnect with old config
+		}
+	}
+}
+
+void StationClass::staticWpsConfigCallback(wps_cb_status status) {
+	WifiStation.internalWpsConfig(status);
+}
+
+bool StationClass::wpsConfigStart(WPSConfigDelegate callback) {
+	debugf("WPS start\n");
+	wpsConfigCallback=callback;
+	wifi_station_disconnect();
+	wifi_set_opmode_current(STATION_MODE);
```

Comment:
If the device has AP enabled either using the SOFTAP_MODE or the STATIONAP_MODE It would be better to replace

`wifi_set_opmode_current(STATION_MODE);`

with 

`wifi_set_opmode_current(wifi_get_opmode() | STATION_MODE);`

That small change will keep the AP mode enabled.